packages/near-membrane-base/src/environment.ts

```diff
@@ -49,39 +52,37 @@ export class VirtualEnvironment {
 
     private redCallableDefineProperty: CallableDefineProperty;
 
-    constructor(options: VirtualEnvironmentOptions) {
-        if (options === undefined) {
+    constructor(providedOptions: VirtualEnvironmentOptions) {
+        if (providedOptions === undefined) {
             throw new ErrorCtor(`Missing VirtualEnvironmentOptions options bag.`);
         }
-        const { blueConnector: localInit, redConnector: foreignInit, distortionCallback } = options;
+        // eslint-disable-next-line prefer-object-spread
+        const options = ObjectAssign({ __proto__: null }, providedOptions);
+        const {
+            blueConnector: localInit,
+            distortionCallback,
+            redConnector: foreignInit,
+            support,
+        } = options;
 
         let blueHooks: Parameters<HooksCallback>;
         let redHooks: Parameters<HooksCallback>;
 
         const blueExportsCallback: HooksCallback = (...hooks) => {
             blueHooks = hooks;
         };
+        const localConnect = localInit('blue', blueExportsCallback, {
+            distortionCallback,
+            trapMutations: false,
+            support,
+        });
         const redExportsCallback: HooksCallback = (...hooks) => {
             redHooks = hooks;
         };
-
-        const initLocalOptions = {
-            distortionCallback,
-        };
-
-        const localConnect = localInit(
-            undefinedSymbol,
-            'blue',
-            ShouldNotTrapMutation,
-            blueExportsCallback,
-            initLocalOptions
-        );
-        const foreignConnect = foreignInit(
-            undefinedSymbol,
-            'red',
-            ShouldTrapMutation,
-            redExportsCallback
-        );
+        const foreignConnect = foreignInit('red', redExportsCallback, {
+            trapMutations: true,
+            support,
```

Comment:
this will not work in ShadowRealm because init is a callable, meaning you can't pass an object around. This PR is probably a throw away because of that :(

we should definitely call init CallableInit (at least the foreign version of it, the other one will allow objects just fine since it is running in blue.